src/lumigo_tracer/lambda_tracer/lambda_reporter.py

```diff
@@ -275,14 +288,51 @@ def _get_prioritized_spans(
             span_metadata_size = spans_to_send_sizes[index]
             span_size = get_event_base64_size(span)
 
-            if current_size + span_size - span_metadata_size < request_max_size:
+            if current_size + span_size - span_metadata_size < buffered_max_size:
                 spans_to_send_dict[index] = span
                 current_size += span_size - span_metadata_size
             else:
                 # This is an optimization step. If the spans are too big, don't try to send them.
                 too_big_spans += 1
                 if too_big_spans >= too_big_spans_threshold:
                     break
+
+        # If we dropped spans we need to update the enrichment spans dropped spans reasons
+        if len(spans_to_send_dict) != len(msgs):
+            enrichment_spans_indexes = [
+                index
+                for index, span in spans_to_send_dict.items()
+                if span.get("type") == ENRICHMENT_TYPE
+            ]
+            if not enrichment_spans_indexes or len(enrichment_spans_indexes) != 1:
+                # We should never get here, if we did it probably means a bug in the tracer code.
+                get_logger().warning(
+                    f"Got unsupported number of enrichment spans: {len(enrichment_spans_indexes)}"
+                )
+            else:
+                enrichment_spans_index = enrichment_spans_indexes[0]
+                enrichment_span_size_before = get_event_base64_size(msgs[enrichment_spans_index])
+                dropped_spans_reasons = spans_to_send_dict[enrichment_spans_index].get(
+                    DROPPED_SPANS_REASONS_KEY, {}
+                )
+                dropped_spans_reasons[DroppedSpansReasons.SPANS_SENT_SIZE_LIMIT.value] = {
+                    "drops": len(msgs) - len(spans_to_send_dict),
+                }
+                spans_to_send_dict[enrichment_spans_index][
+                    DROPPED_SPANS_REASONS_KEY
+                ] = dropped_spans_reasons
+                enrichment_span_size_after = get_event_base64_size(
+                    spans_to_send_dict[enrichment_spans_index]
+                )
+                size_increase = enrichment_span_size_after - enrichment_span_size_before
+                current_size += size_increase
+                if current_size > buffered_max_size:
+                    # TODO: Remove more spans based on the order until we are within the limit
+                    get_logger().warning(
+                        f"Enrichment span size increased by {size_increase} bytes, "
+                        f"making the total size too big: {current_size} bytes (max: {buffered_max_size} bytes)"
+                    )
```

Comment:
Can we have this in a separate method? something like `update_enrichment_spans`? The `_get_prioritized_spans` is already quite big and getting a little hard to follow what's going on.

Reply:
Done